homeassistant/components/smartthings/fan.py

```diff
@@ -41,19 +41,51 @@ async def async_setup_entry(
 
 def get_capabilities(capabilities: Sequence[str]) -> Sequence[str] | None:
     """Return all capabilities supported if minimum required are present."""
-    supported = [Capability.switch, Capability.fan_speed]
-    # Must have switch and fan_speed
-    if all(capability in capabilities for capability in supported):
-        return supported
-    return None
+
+    # MUST support switch as we need a way to turn it on and off
+    if Capability.switch not in capabilities:
+        return None
+
+    # These are all optional but at least one must be supported
+    optional = [
+        Capability.air_conditioner_fan_mode,
+        Capability.fan_speed,
+    ]
+
+    # If none of the optional capabilities are supported then error
+    if not any(capability in capabilities for capability in optional):
+        return None
+
+    supported = [Capability.switch]
+
+    for capability in optional:
+        if capability in capabilities:
+            supported.append(capability)
+
+    return supported
 
 
 class SmartThingsFan(SmartThingsEntity, FanEntity):
     """Define a SmartThings Fan."""
 
-    _attr_supported_features = FanEntityFeature.SET_SPEED
+    _attr_supported_features = FanEntityFeature(0)
     _attr_speed_count = int_states_in_range(SPEED_RANGE)
 
+    def __init__(self, device):
+        """Init the class."""
+        super().__init__(device)
+        self._attr_supported_features = self._determine_features()
+
+    def _determine_features(self):
+        flags = FanEntityFeature(0)
```

Comment:
Use the enums and not int's

Reply:
`FanEntityFeature(0)` initializes `flags` to none with no features. This is how this is used in the fan module https://github.com/home-assistant/core/blob/dev/homeassistant/components/fan/__init__.py#L239. As this is a temp variable it is being initialized to none and then filled based on the capabilities of the device.